src/modules/job-manager/queue.c

```diff
@@ -270,6 +294,15 @@ int queue_restore_state (struct queue *queue, int version, json_t *o)
             errno = EINVAL;
             return -1;
         }
+        if (version == 1) {
+            if (json_unpack (entry,
+                             "{s:b s?s}",
+                             "start", &start,
+                             "stop_reason", &stop_reason) < 0) {
+                errno = EINVAL;
+                return -1;
+            }
+        }
```

Comment:
Suggestion: don't share code between versions.  Have one unpack for all of version 0, and one unpack for all of version 1.  The fact that we have a version number means that there doesn't have to be any compatibility at all between versions, so assuming that in the code doesn't telegraph the design properly IMHO.